### PR TITLE
Set .so version number to 1.0.0

### PR DESCRIPTION
**Pull request by saraedum**
also set the .so number to 1.0.0 (which is not semantic versioning but
now just happens to be the same.)

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -5,6 +5,8 @@ AM_CFLAGS = $(OPENMP_CFLAGS)
 
 # Installed library (libtool (LT) libraries)
 lib_LTLIBRARIES = libeantic.la libeanticxx.la
+libeantic_la_LDFLAGS = -version-info @libeantic_version_info@ -no-undefined
+libeanticxx_la_LDFLAGS = -version-info @libeanticxx_version_info@ -no-undefined
```
Comment (saraedum) [inline Makefile.am]: `-no-undefined` helps so we know that we did not forget to implement a method that we declared in a class.

Comment (saraedum) [reply Makefile.am]: Actually, that's not true. It just makes sure that nothing that was there at compile time is missing at link time.

File: e-antic/poly_extra.h [hunk 2]
```diff
@@ -322,7 +322,6 @@ fmpz_poly_resultant_modular_div(fmpz_t res, const fmpz_poly_t poly1,
 slong _fmpz_poly_remove_content_2exp(fmpz * pol, slong len);
 void _fmpz_poly_scale_2exp(fmpz * pol, slong len, slong k);
 
-int fmpq_set_str(fmpq_t x, const char *str, int base);
```
Comment (videlec) [inline e-antic/poly_extra.h]: Why this removal!?

**Comment by videlec** [comment]
Travis reports
```
+ ./configure --prefix=/usr --enable-valgrind
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
```

File: e-antic/poly_extra.h (hunk 2 above)
Comment (saraedum) [reply e-antic/poly_extra.h]: Sorry, should have added a comment. This is a duplicate. (Line 302.)

File: e-antic/poly_extra.h [hunk 3]
```diff
@@ -335,8 +334,6 @@ void _fmpq_vec_sort(fmpq * vec, slong len);
 void _fmpq_vec_randtest_uniq_sorted(fmpq * vec, flint_rand_t state, slong len, mp_bitcnt_t bits);
 void fmpz_poly_randtest_no_real_root(fmpz_poly_t p, flint_rand_t state,
                                                 slong len, mp_bitcnt_t bits);
- void _fmpz_poly_num_real_roots_sturm(slong * n_neg, slong * n_pos, const fmpz * pol, slong len);
```
Comment (saraedum) [inline e-antic/poly_extra.h]: Same here (line 313.)

**Comment by saraedum** [comment]
That's odd. I rebased on top of master. Let's see if that makes a difference.

**Comment by videlec** [comment]
It does :-)

**Pull request merged**